presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationCompiler.java

```diff
@@ -254,4 +230,209 @@ private static Method getCombineFunction(Class<?> clazz, Class<?> stateClass)
 
         return stateClasses;
     }
+
+    private static String getDescription(AnnotatedElement base, AnnotatedElement override)
+    {
+        Description description = override.getAnnotation(Description.class);
+        if (description != null) {
+            return description.value();
+        }
+        description = base.getAnnotation(Description.class);
+        return (description == null) ? null : description.value();
+    }
+
+    private static Set<String> getLiteralParameter(Method inputFunction)
+    {
+        ImmutableSet.Builder<String> literalParametersBuilder = ImmutableSet.builder();
+
+        Annotation[] literalParameters = inputFunction.getAnnotations();
+        for (Annotation annotation : literalParameters) {
+            if (annotation instanceof LiteralParameters) {
+                for (String literal : ((LiteralParameters) annotation).value()) {
+                   literalParametersBuilder.add(literal);
+                }
+            }
+        }
+
+        return literalParametersBuilder.build();
+    }
+
+    private static List<Method> findPublicStaticMethodsWithAnnotation(Class<?> clazz, Class<?> annotationClass)
+    {
+        ImmutableList.Builder<Method> methods = ImmutableList.builder();
+        for (Method method : clazz.getMethods()) {
+            for (Annotation annotation : method.getAnnotations()) {
+                if (annotationClass.isInstance(annotation)) {
+                    checkArgument(Modifier.isStatic(method.getModifiers()) && Modifier.isPublic(method.getModifiers()), "%s annotated with %s must be static and public", method.getName(), annotationClass.getSimpleName());
+                    methods.add(method);
+                }
+            }
+        }
+        return methods.build();
+    }
+
+    public static class BindableAggregationFunction
+    {
+        private final Signature signature;
+        private final String description;
+        private final boolean approximate;
+        private final boolean decomposable;
+
+        private final Class<?> definitionClass;
+        private final Class<?> stateClass;
+        private final Method inputFunction;
+        private final Method outputFunction;
+
+        public BindableAggregationFunction(Signature signature,
```

Comment:
If you split parameters then put each parameter in new line. This one of the comments I got the most frequently from Martin.